### PR TITLE
74619 - ess api additional params calculation when aio feature on

### PR DESCRIPTION
**Pull request by MastekRohitChettri**
- Added code changes
- Added unit test cases
- Added functional tests

This PBI calculate the additional AIO params as below,

- requestedAioProductCount
- aioExchangeSetCellCount
- requestedAioProductsAlreadyUpToDateCount

Note :- Excluded config, response and helpers files from code coverage to improve Stryker check report.

### CODE REVIEW

**Review by ray-sj — APPROVED** [review]
looks fine

**Review by DanOwenUKHO — CHANGES_REQUESTED** [review]
A few questions to address

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Services/ProductDataServiceTests.cs [hunk 1]
```diff
@@ -127,35 +127,36 @@ private ExchangeSetResponse GetExchangeSetResponse()
             return exchangeSetResponse;
         }
 
-        private ExchangeSetResponse GetExchangeSetResponseAioToggleOff()
+        private static ExchangeSetResponse GetExchangeSetResponseAioToggleOff()
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Services/ProductDataServiceTests.cs:130]: Made private methods static as suggested by dot net compiler to achieve performance gain.
Additionally, these methods can't be use outside so no new instance will require.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Services/ProductDataServiceTests.cs [hunk 2]
```diff
@@ -178,25 +179,30 @@ private ExchangeSetResponse GetExchangeSetResponseAioToggleOff()
             return exchangeSetResponse;
         }
 
-        private ExchangeSetResponse GetExchangeSetResponseAioToggleON()
+        private static ExchangeSetResponse GetExchangeSetResponseAioToggleON()
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Services/ProductDataServiceTests.cs:182]: Made private methods static as suggested by dot net compiler to achieve performance gain.
Additionally, these methods can't be use outside so no new instance will requires.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Services/ProductDataServiceTests.cs [hunk 3]
```diff
@@ -264,7 +270,7 @@ private SalesCatalogueResponse GetSalesCatalogueResponse()
                 ScsRequestDateTime = DateTime.UtcNow
             };
         }
-        private SalesCatalogueResponse GetSalesCatalogueFileSizeResponse()
+        private static SalesCatalogueResponse GetSalesCatalogueFileSizeResponse()
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Services/ProductDataServiceTests.cs:273]: Made private methods static as suggested by dot net compiler to achieve performance gain.
Additionally, these methods can't be use outside so no new instance will requires.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Services/ProductDataService.cs [hunk 4]
```diff
@@ -415,7 +336,7 @@ private ExchangeSetResponse MapExchangeSetResponse(SalesCatalogueResponse salesC
             return model;
         }
 
-        private string ConvertLastModifiedToString(SalesCatalogueResponse salesCatalougeResponse)
+        private static string ConvertLastModifiedToString(SalesCatalogueResponse salesCatalougeResponse)
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Services/ProductDataService.cs:339]: Made private methods static as suggested by dot net compiler to achieve performance gain.
Additionally, these methods can't be use outside so no new instance will requires.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Services/ProductDataService.cs [hunk 5]
```diff
@@ -243,6 +189,11 @@ public async Task<ExchangeSetServiceResponse> CreateProductDataByProductVersions
                 salesCatalogueResponse.ResponseBody.ProductCounts.RequestedProductsNotReturned = new List<RequestedProductsNotReturned>();
                 salesCatalogueResponse.ResponseBody.ProductCounts.RequestedProductCount = salesCatalogueResponse.ResponseBody.ProductCounts.RequestedProductsAlreadyUpToDateCount = request.ProductVersions.Count;
             }
+            else if (salesCatalogueResponse.ResponseCode == HttpStatusCode.OK) //Set Aio details on exchange set response
+            {
+                IEnumerable<string> lstRequestedProducts = request.ProductVersions.Select(x => x.ProductName).ToList();
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Services/ProductDataService.cs]: Additional conversion has been removed.

**Review by shailen14599 — APPROVED** [review]
LGTM

**Comment by DanOwenUKHO** [comment]
Stryker report ->  44.97 %

**Review by DanOwenUKHO — APPROVED** [review]
LGTM

**Pull request merged**